src/client/content_mapblock.cpp

```diff
@@ -1377,6 +1377,47 @@ void MapblockMeshGenerator::drawNodeboxNode()
 
 	std::vector<aabb3f> boxes;
 	n.getNodeBoxes(nodedef, &boxes, neighbors_set);
+
+	bool isTransparent = false;
+
+	for (const TileSpec &tile : tiles) {
+		if (tile.layers[0].isTransparent()) {
+			isTransparent = true;
+			break;
+		}
+	}
+
+	if (isTransparent) {
+		for (int axis = 0; axis < 3; axis++) {
+			// identify sections
+			std::vector<float> sections;
```

Comment:
this can be optimized regarding allocations like so:
```suggestion
		std::vector<float> sections;
		sections.reserve(8 + 2 * boxes.size());
		for (int axis = 0; axis < 3; axis++) {
			// identify sections
			sections.clear();
```